packages/fxa-settings/src/models/integrations/oauth-integration.ts

```diff
@@ -45,64 +57,110 @@ export function isOAuthIntegration(integration: {
 
 // TODO: probably move this somewhere else
 export class OAuthIntegrationData extends BaseIntegrationData {
-  @bind([V.isString], T.snakeCase)
+  @IsOptional()
+  @IsHexadecimal()
+  @bind(T.snakeCase)
   clientId: string | undefined;
```

Comment:
Something I noticed recently is in the backbone version, if client_id is present, I think we always try to fetch/validate it immediately. If you go to an RP in staging and change `client_id`, it immediately shows an "invalid param client_id" error, and in the React version we don't immediately error and probably should. Not sure if we can read from the config here to check against valid client_ids but feels like that's another task, maybe FXA-8106 or FXA-6836, but if you're going to file a follow up for the other "TODO - Validation" comments I could see us looking at it then.

Reply:
That is interesting case. Thanks for call it out. I think we probably could do this dynamically. All though so far all the validations are statically defined. I've added the `TODO - Validation` comment here.